Sources/SwiftGodotMacroLibrary/Extensions/TypeSyntax+MacroExport.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  TypeSyntax+MacroExport.swift
+//  
+//
+//  Created by Estevan Hernandez on 11/22/23.
+//
+
+import SwiftSyntax
+
+extension TypeSyntax {
+	/// Returns `true` if type is either `[Element]` or `Array<Element>`
+	var isArray: Bool {
+		isSquareArray || isGenericArray
+	}
+	
+	var isVariantCollection: Bool {
+		self.as(IdentifierTypeSyntax.self)?.name.text == "VariantCollection"
+	}
+	
+	var variantCollectionElementTypeName: String? {
```

Comment:
TODO: this will not parse the following, because it assumes the type is on the identifier type syntax, but in this case it would be on the initializer

```swift
let variantCollection = VariantCollection<Node>()
```